packages/tools/examples/segmentBidirectionalTool/index.ts

```diff
@@ -0,0 +1,465 @@
+import {
+  RenderingEngine,
+  Types,
+  Enums,
+  setVolumesForViewports,
+  volumeLoader,
+  getEnabledElement,
+} from '@cornerstonejs/core';
+import * as cornerstoneTools from '@cornerstonejs/tools';
+
+import {
+  initDemo,
+  createImageIdsAndCacheMetaData,
+  setTitleAndDescription,
+  addDropdownToToolbar,
+  addSliderToToolbar,
+  setCtTransferFunctionForVolumeActor,
+  addButtonToToolbar,
+} from '../../../../utils/demo/helpers';
+
+// This is for debugging purposes
+console.warn(
+  'Click on index.ts to open source code for this example --------->'
+);
+
+const {
+  SegmentationDisplayTool,
+  ToolGroupManager,
+  Enums: csToolsEnums,
+  segmentation,
+  RectangleScissorsTool,
+  SphereScissorsTool,
+  CircleScissorsTool,
+  BrushTool,
+  PaintFillTool,
+  PanTool,
+  ZoomTool,
+  StackScrollTool,
+  StackScrollMouseWheelTool,
+  BidirectionalTool,
+  utilities: cstUtils,
+} = cornerstoneTools;
+
+const { roundNumber } = cstUtils;
+
+const { MouseBindings, KeyboardBindings } = csToolsEnums;
+const { ViewportType } = Enums;
+const { segmentation: segmentationUtils } = cstUtils;
+let renderingEngine;
+const viewportId = 'CT_AXIAL';
+
+// Define a unique id for the volume
+const volumeName = 'CT_VOLUME_ID'; // Id of the volume less loader prefix
+const volumeLoaderScheme = 'cornerstoneStreamingImageVolume'; // Loader id which defines which volume loader to use
+const volumeId = `${volumeLoaderScheme}:${volumeName}`; // VolumeId with loader id + volume id
+const segmentationId = 'MY_SEGMENTATION_ID';
+const toolGroupId = 'MY_TOOLGROUP_ID';
+let representationUID;
+
+const actionConfiguration = {
+  contourBidirectional: {
+    method: cstUtils.segmentation.segmentContourAction,
+    bindings: [
+      {
+        key: 'c',
+      },
+    ],
+    segmentData: new Map(),
```

Comment:
can you go a level deeper, so that we have an architecture here? so instead of what you have how about

```
const actionConfiguration = {
  contourBidirectional: {
    method: cstUtils.segmentation.segmentContourAction,
    bindings: [
      {
        key: 'c',
      },
    ],
    data: {
      segmentData: new Map(),
      // later we can put more things here or other actions can put more here
    }
  },
};
```